### PR TITLE
Addressing issue #16

### PR DESCRIPTION
**Pull request by LucaCappelletti94**
With this pull request, I fix the issue #16 and fix a couple of code smells.

### CODE REVIEW

File: src/bit_vector.rs [hunk 1]
```diff
@@ -168,6 +168,10 @@ impl FID for BitVector {
         self.len
     }
 
+    fn is_empty(&self) -> bool {
```
Comment (ajalab) [inline src/bit_vector.rs]: Could you move this to `FID` trait definition as a default method? We can keep `BitVector` implementation minimal then

**Review by ajalab — CHANGES_REQUESTED** [review]
Thank you for working on this!

File: src/bit_vector.rs (hunk 1 above)
Comment (LucaCappelletti94) [reply src/bit_vector.rs]: Done in commit #02e5742.

**Review by ajalab — APPROVED** [review]
Thanks again!

**Pull request merged**